controllers/validation/account_validation_controller.go

```diff
@@ -450,51 +451,58 @@ func (r *AccountValidationReconciler) Reconcile(ctx context.Context, request ctr
 
 	// check if account belongs to accountpool
 	if !account.IsBYOC() {
-		awsRegion := config.GetDefaultRegion()
-		awsSetupClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
-			SecretName: utils.AwsSecretName,
-			NameSpace:  awsv1alpha1.AccountCrNamespace,
-			AwsRegion:  awsRegion,
-		})
+		result, err := r.ValidateRegionalServiceQuotas(reqLogger, &account, r.awsClientBuilder)
 		if err != nil {
-			connErr := fmt.Sprintf("unable to connect to default region %s", awsRegion)
-			reqLogger.Error(err, connErr)
-			return utils.RequeueWithError(err)
+			return utils.DoNotRequeue()
```

Comment:
I think this changes the logic of the code a lot.
The old code forced a requeue by returning an error result, but the new code never requeues when an error is returned - is this really required/wanted?

Reply:
@bergmannf  yeah I think its best to have not re-queue if there is an error. I'll change it to just return `reconcile.Result{}, err`  to eliminate any confusion as to the expected result

Reply:
I think it should explicitly requeue when there is an error. In order to clean up the confusion around this part of the code, i'd would rather change `ValidateRegionalServiceQuotas` to just return an error, ideally an `AccountValidationError` as it's done with the other validation functions.

Reply:
And to clarify this a bit:
Currently, all the decision if an account should be requeued are all handled in the reconcile loop. The individual `Validate` functions just return a custom error type when the validation didn't succeed and the Reconcile() function then decides if a requeue should happen in that case or not.
I think it would be good to stick to that pattern to keep the responsibilities clear and so we have a single function that controls the requeing of the accounts, and not have that logic spread all over the place.

In this case, instead of returning `reconcile.Result{RequeueAfter: 10 * time.Minute}, nil` for example, the function would return:
```go
return &AccountValidationError{
	Type: NotAllServicequotasApplied, //<- new ValidationError
	Err:  errors.New("Service Quotas not yet applied"),
}
```
And in the Recocile loop it would then check the error code and requeue the account as appropriate. See the `ValidateAccountTags` function for example.